_practicalities/intro.md

```diff
@@ -0,0 +1,355 @@
+planning and communication, letting your user base know in advance that the
+transition is happening and what step to take is critical for a transition.
+
+For your library code here the steps you need to take to ensure that
+installation will fail in the least number of case:
+
+You need to release your new packages version with
+[setuptools](https://pypi.python.org/pypi/setuptools) version 24.2.0 or above.
+You can also use one of the alternate package manager that can set the
+[Requires-Python](https://www.python.org/dev/peps/pep-0345/#requires-python)
+metadata field. Without this, pip 9.0 **will try** to install non-compatible
+version of your software on Python 2. This version of setuptools is recent
+(July 20, 2016) and this possible thank to the [work of Xavier
+Fernandez](https://github.com/pypa/setuptools/pull/631)
+
+Add the following to your `setup.py`
+
+```
+setup(
+   ...
+   python_requires='>=3.3'
+   ...
+)
+```
+
+Changes `>=3.3` accordingly depending on what version your library decides to
+support. In particular you can use `>=2.6` or `>=3.5` ! Note that this also
+support the _compable with_ syntax: `~=2.5` (meaning, `>=2.5` and `<3`.
+
+This will make [PyPI aware](https://github.com/pypa/warehouse/pull/1448) that
+your package is Python 3.3+ only, and [allow
+pip](https://github.com/pypa/pip/pull/3877) to be [made aware of
+this](https://github.com/pypa/pypi-legacy/pull/506).
+
+Thus as long as your user have a recent enough version of pip, and setuptools
+they will get the right version of your library.
+
+# Unit Testing and documentation
+
+It is recommended **not** to invoke `setup.py` directly either with `install` or
+`develop` subcommands. These may not correctly resolve dependencies, and can
+install incompatible versions of dependencies. Please recommend and use `pip
+install . `  and `pip install -e .` for regular and developer install.
+
+Check in scripts, and documentation that the correct installation command is
+used. 
+
+# Recommended Mitigations
+
+These are not mandatory but should make the transition seamless by warning your
+user early enough _and_ providing useful error messages.
+
+## Runtime warning on master
+
+Add a warning at _runtime_ early on master (before switching to Python 3
+  only)
+
+```
+import warnings
+import sys
+if sys.version_info < (3,):
+    warnings.warn('You are using master of `Frobulator` with Python 2. '
+                  'Frobulator will soon be Python 3 only. '
+                  'See this issue to know more.',
+                  UserWarning)
+```
+
+Your Python 2 user have a chance to upgrade, or get off master, (for example on
+the LTS branch).
+
+## Fail early at import time
+
+Add an error early at import at runtime with a clear error message, leave the
+early import compatible Python 2 for users to not be welcomed with a useless
+`SyntaxError`. You are _allowed_ to use multi-line strings in error messages.
```

Comment:
Emphasising _allowed_ seems odd here. Does that mean it's allowed but discouraged?